packages/venia-ui/lib/components/CartPage/ProductListing/EditModal/productForm.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import gql from 'graphql-tag';
+import { Form } from 'informed';
+import { useProductForm } from '@magento/peregrine/lib/talons/CartPage/ProductListing/EditModal/useProductForm';
+
+import { mergeClasses } from '../../../../classify';
+import Button from '../../../Button';
+import LoadingIndicator from '../../../LoadingIndicator';
+import Options from '../../../ProductOptions';
+import { QuantityFields } from '../quantity';
+import defaultClasses from './productForm.css';
+import { CartPageFragment } from '../../cartPageFragments';
+
+const ProductForm = props => {
+    const { item: cartItem, setIsUpdating } = props;
+    const talonProps = useProductForm({
+        cartItem,
+        getConfigurableOptionsQuery: GET_CONFIGURABLE_OPTIONS,
+        setIsUpdating,
+        updateConfigurableOptionsMutation: UPDATE_CONFIGURABLE_OPTIONS_MUTATION,
+        updateQuantityMutation: UPDATE_QUANTITY_MUTATION
+    });
+    const {
+        configItem,
+        handleOptionSelection,
+        handleSubmit,
+        isLoading,
+        setFormApi
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    if (isLoading) {
+        return (
+            <LoadingIndicator
```

Comment:
I believe we can achieve some optimization by moving the loading indicator component instance to a `useMemo` and returning the result if `isLoading`. Since only 2 states exist for the component either loading or not potentially only creating 2 instances.

```
....

const loadingIndicator = useMemo(() => {
  if (isLoading) {
     return (
          <LoadingIndicator
               classes={{ root: classes.loading }}
          >{`Fetching Product Options...`}</LoadingIndicator>
     );
  } else {
   return null;
  }
}, [isLoading, classes.loading]);

if (isLoading) { return loadingIndicator }

....
```

Reply:
I remember @jimbo left some comments on my PR about `useMemo`. Don't do this yet. Let me check with Jimmy first.

Reply:
Very interested in this discussion. I have not been a huge fan of our overuse of `useMemo` and `useCallback`, so would be nice to define some usage criteria. Relevant article:

https://kentcdodds.com/blog/usememo-and-usecallback